.github/workflows/compilation_on_macos.yml

```diff
@@ -210,7 +210,7 @@ jobs:
             #$AOT_BUILD_OPTIONS,
           ]
         os: [macos-latest]
-        wasi_sdk_release: ["https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-12/wasi-sdk-12.0-macos.tar.gz"]
+        wasi_sdk_release: ["https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-17/wasi-sdk-17.0-macos.tar.gz"]
```

Comment:
Wonder if we want to continue building for 12 to ensure backward compatibility? Or do we want to drop 12 and ask users to migrate to >= 17 now?

Reply:
I tried `wasi-sdk` version 12 and `wasi-libc` branch `wasi-sdk-17` [here](https://github.com/eloparco/wasm-micro-runtime/actions/runs/3730554354/jobs/6327794669) but the build failed.
It was not immediate to find a compatible combination of `wasi-sdk` and `wasi-libc` for the different targets (on ubuntu I'm using `wasi-sdk` version 16 in this PR since version 17 was not working).

Reply:
I am a little confused here. It seems the CI depends on the combination of `wasi-sdk-X` + a specific release of `wasi-libc` now. Is there a plan that a future wasi-sdk release include everything and make the local combination not necessary?